processing/src/main/java/io/druid/query/lookup/LookupBean.java

```diff
@@ -19,32 +19,76 @@
 
 package io.druid.query.lookup;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
 
 //TODO merge this code to the same definition when pr/1576 is merged
 class LookupBean
 {
+  private final LookupExtractorFactoryContainer container;
+  private final String name;
+
+  @JsonCreator
+  public LookupBean(
+      @JsonProperty("name") String name,
+      //kept for backward compatibility with druid ver <= 0.9.2 persisted snapshots
+      @Deprecated @JsonProperty("factory") LookupExtractorFactory factory,
+      @JsonProperty("container") LookupExtractorFactoryContainer container
+  )
+  {
+    Preconditions.checkArgument(factory == null || container == null, "only one of factory or container should exist");
+    Preconditions.checkArgument(factory != null || container != null, "one of factory or container must exist");
+
+    this.name = name;
+    this.container = container != null ? container : new LookupExtractorFactoryContainer(null, factory);
+  }
+
   @JsonProperty
-  LookupExtractorFactory factory;
+  public String getName()
+  {
+    return name;
+  }
+
   @JsonProperty
-  String name;
+  public LookupExtractorFactoryContainer getContainer()
+  {
+    return container;
+  }
+
+  @Override
+  public String toString()
+  {
+    return "LookupBean{" +
+           "container=" + container +
+           ", name='" + name + '\'' +
+           '}';
+  }
 
   @Override
   public boolean equals(Object o)
   {
     if (this == o) {
       return true;
     }
-    if (!(o instanceof LookupBean)) {
```

Comment:
these equals/hashcode are generated by IDE and newly generated one looks ok to me. the change is superficial